src/minerd.cpp

```diff
@@ -94,6 +94,15 @@ int main(int argc, char** argv)
         return 1;
     }
 
+    // Validate input GPU device indexes
+    auto info = merit::gpus_info();
+    for(const auto& device: gpu_devices){
+        if(device >= info.size()){
```

Comment:
Is it guaranteed that the indices are continuous?  If so this looks good.

Reply:
yes